pstoremanager: flaky TestClose

```
  === CONT  TestClose
      pstoremanager_test.go:111: Hit timeout
      controller.go:269: missing call(s) to *pstoremanager_test.MockPeerstore.RemovePeer(is equal to t1Zv2yaZ (peer.ID)) /Users/runner/work/go-libp2p/go-libp2p/p2p/host/pstoremanager/pstoremanager_test.go:106
      controller.go:269: aborting test due to missing call(s)
  --- FAIL: TestClose (5.14s)
```

--- a/p2p/host/pstoremanager/pstoremanager_test.go
+++ b/p2p/host/pstoremanager/pstoremanager_test.go
@@ -74,7 +74,6 @@ func TestReconnect(t *testing.T) {
 }
 
 func TestClose(t *testing.T) {
-	t.Parallel()
 	ctrl := gomock.NewController(t)
 	defer ctrl.Finish()
 	eventBus := eventbus.NewBus()
@@ -98,6 +97,7 @@ func TestClose(t *testing.T) {
 	// make sure the event is sent before we close
 	select {
 	case <-sub.Out():
+		time.Sleep(100 * time.Millisecond) // make sure this event is also picked up by the pstoremanager
 	case <-time.After(5 * time.Second):
 		t.Fatalf("Hit timeout")
 	}

